From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Morgan Hough <mhough@fedoraproject.org>
Date: Fri, 13 Mar 2026 12:00:00 +0000
Subject: [PATCH 1/2] Increase sleep for tests

Increase sleep times in network and TLS tests to avoid sporadic failures
on slow build systems. Updated for 3.7.0 where dcmtls tests use
OFStandard::forceSleep() instead of force_sleep().
---
 dcmnet/tests/tscusession.cc | 6 +++---
 dcmtls/tests/tscuscptls.cc  | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/dcmnet/tests/tscusession.cc b/dcmnet/tests/tscusession.cc
index 000000000..000000001 100644
--- a/dcmnet/tests/tscusession.cc
+++ b/dcmnet/tests/tscusession.cc
@@ -164,7 +164,7 @@ void configure_scp_for_sop_class(DcmSCPConfig& cfg,
 void scu_session(const OFString& called_ae_title, const OFString& sop_class)
 {
     // Make sure server is up
-    force_sleep(1);
+    force_sleep(10);
     // Basic configuration
     SessionSCU scu;
     scu.setAETitle("TEST_SCU");
@@ -195,13 +195,13 @@ OFTEST_FLAGS(dcmnet_scu_session_handler, EF_Slow)

     // Send FIND, and wait to be sure SCP has time to exit
     scu_session("FIND_SESSION", UID_FINDPatientRootQueryRetrieveInformationModel);
-    force_sleep(1);
+    force_sleep(10);
     scp.join();

     // Send MOVE, and wait to be sure SCP has time to exit
     scp.start();
     scu_session("MOVE_SESSION", UID_MOVEPatientRootQueryRetrieveInformationModel);
-    force_sleep(1);
+    force_sleep(10);
     scp.join();

     // TODO: Test C-GET
diff --git a/dcmtls/tests/tscuscptls.cc b/dcmtls/tests/tscuscptls.cc
index 000000000..000000001 100644
--- a/dcmtls/tests/tscuscptls.cc
+++ b/dcmtls/tests/tscuscptls.cc
@@ -327,7 +327,7 @@ OFTEST_FLAGS(dcmtls_scp_tls, EF_None)
       port_number = 0xF000 + (rnd.getRND16() & 0xFFF);
       config.setPort(port_number);
       scp.start();
-      OFStandard::forceSleep(2); // wait 2 seconds for the SCP process to start
+      OFStandard::forceSleep(10); // wait 10 seconds for the SCP process to start
       memory_barrier.lock();
       memory_barrier.unlock();
     }
@@ -407,7 +407,7 @@ OFTEST_FLAGS(dcmtls_scp_pool_tls, EF_None)
       port_number = 0xF000 + (rnd.getRND16() & 0xFFF);
       config.setPort(port_number);
       pool.start();
-      OFStandard::forceSleep(2); // wait 2 seconds for the SCP process to start
+      OFStandard::forceSleep(10); // wait 10 seconds for the SCP process to start
     }
     while ((i++ < 5) && (! pool.m_is_running)); // try up to 5 port numbers before giving up
     if (! pool.m_is_running) BAILOUT("Start of the SCP thread pool failed: " << pool.m_listen_result.text());
@@ -435,7 +435,7 @@ OFTEST_FLAGS(dcmtls_scp_pool_tls, EF_None)

     // "ensure" the pool is initialized before any SCU starts connecting to it. The initialization
     // can take a couple of seconds on older systems, e.g. debian i368.
-    OFStandard::forceSleep(5);
+    OFStandard::forceSleep(15);

     for (OFVector<TestTLSSCU*>::const_iterator it2 = scus.begin(); it2 != scus.end(); ++it2)
     {
--
2.48.1

